src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -4503,6 +4503,14 @@ bool Unit::HasAuraTypeWithValue(AuraType auraType, int32 value) const
     return false;
 }
 
+bool Unit::HasAuraTypeWithTriggerSpell(AuraType auratype, uint32 triggerSpell) const
+{
+    for (auto&& aura : GetAuraEffectsByType(auratype))
```

Comment:
no auto for value iteration

Reply:
wait, if i remember right, auto is allowed for iterators

Reply:
`auto` is allowed for iterators when you actually have the iterator object (so when you use `.begin()` and `.end()`, not in range style loop)

Reply:
Like this?

`for (AuraEffect const* aura : GetAuraEffectsByType(auratype))`

Reply:
Yes

Reply:
Done